CNCF page

attfarhan opened the pull request:
Addresses https://github.com/sourcegraph/sourcegraph/issues/14318. 

We want to add a community page for Sourcegraph/cncf. This searches over the [repogroup cncf](https://sourcegraph.com/search?q=repogroup:cncf&patternType=literal).

- [x]  Include all CNCF projects in repogroup:cncf [commit](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/commit/5d7dce1a56e799c6b8167ee58c2c68ac25c67ee1)
- [x]  Add CNCF repogroup on the Sourcegraph.com homepage (bump to first repogroup)
- [x]  This opens to a clean homepage (http://sourcegraph.com/cncf) with 
       - CNCF logo 
       - a title right below (Search the 67 CNCF projects -> linking to the list of [projects](https://landscape.cncf.io/project=hosted)). Note this looks like a **clean** homepage (don't show Example searches or any other sections below the search bar)
- [x]  In the Search bar, omit '**repogroup:**', it's fine that repogroup: will appear when you go to the search results page
- [x]  Remove everything from the top navbar (preferable both when signed it or not) - campaigns, extensions, code insights (only visible if feature flag is enabled), etc.
- [x] Any code host, browser extensions, etc. toasts and notifications are removed. **Is there anything else obtrusive we can remove?**

Out of scope for this PR:
- [ ]  Remove the cookie notice (needs to happen in cookiebot)
- [ ] Set up opsgenie monitoring and logging for usage => https://github.com/sourcegraph/deploy-sourcegraph-dot-com/pull/3562
- [ ] LSIF code-intel for as many repos as possible (mostly Go?)
- [ ] User test this!
- [ ]  Ensure CNCF repositories are cloned and indexed

limitedmage commented on web/src/search/input/SearchPageInput.tsx:
```diff
@@ -236,7 +238,9 @@ export const SearchPageInput: React.FunctionComponent<Props> = (props: Props) =>
             submitSearch({
                 ...props,
-                query: userQueryState.query,
+                query: props.hiddenQuerySuffix
```
Can we add some unit tests for this new functionality?

limitedmage commented on web/src/routes.tsx:
```diff
@@ -212,6 +213,11 @@ export const routes: readonly LayoutRouteProps<any>[] = [
     },
+    {
+        path: '/cncf',
+        render: props => <RepogroupPage {...props} repogroupMetadata={cncf} lowProfile={true} />,
```
Can we make `lowProfile` part of the repogroup metadata instead of hardcoding it here just for CNCF?

limitedmage commented on web/src/repogroups/cncf.tsx:
```diff
@@ -0,0 +1,20 @@
+        <>
+            Search the{' '}
+            <a href="https://landscape.cncf.io/project=hosted" target="_blank" rel="noopener noreferrer">
+                67 CNCF projects
```
I'm concerned that hardcoding the number 67 here will make it harder to simply update the list in the config.
Additionally, do we want to use CNCF's full name in the description?

rvantonder replied:
+1 for expanding acronyms, at least once on this page. I think it's appropriate in the Title `Cloud Native Computing Foundation (CNCF)`

poojaj-tech replied:
Agreed. Expanding to the full name in the title makes sense. cc: @rrhyne any thoughts on the name and length?

It would be great to mirror their website. The website uses inconsistent naming though.
1. Cloud Native Interactive Landscape - Main page (https://www.cncf.io/)
2. CNCF Cloud Native Interactive Landscape (the page we link to - https://landscape.cncf.io/project=hosted)

attfarhan replied:
So what should the description be @poojaj-tech? Should we just remove the number?

limitedmage commented on web/src/repogroups/RepogroupPage.tsx:
```diff
@@ -58,6 +58,7 @@ export interface RepogroupPageProps
 
     /** Whether globbing is enabled for filters. */
     globbing: boolean
+    lowProfile?: boolean
```
Can we add a storybook test for the reprogroup page in low profile mode?

poojaj-tech commented:
@attfarhan took this for a spin - looks really awesome and clean  🎉


Feedback:

1. 2 more things to remove - Query builder and Version Contexts.
<img width="820" alt="Screen Shot 2020-10-01 at 2 56 29 PM" src="https://user-images.githubusercontent.com/57463526/94854977-c9aa7c00-03fb-11eb-9589-75fea85d648f.png">
2.  On the search results page, we should add **repogroup:cncf** to the beginning of the query instead of the end of it to signify the user is searching within this scope, even though the order doesn't matter.
<img width="1273" alt="Screen Shot 2020-10-01 at 3 12 45 PM" src="https://user-images.githubusercontent.com/57463526/94855057-eb0b6800-03fb-11eb-9546-bc2eaee06f4b.png">
3.  I noticed the position of the page title (CNCF) (and the titles on other repogroup pages) is slightly moved up. Moving it down by some pixels to match sourcegraph.com/search homepage would center it closer to a homepage experience and make the space below the title feel less empty.

poojaj-tech commented on web/src/repogroups/cncf.tsx:
```diff
@@ -0,0 +1,20 @@
+    description: (
+        <>
+            Search the{' '}
+            <a href="https://landscape.cncf.io/project=hosted" target="_blank" rel="noopener noreferrer">
```
@daxmc99 is this a good link to surface the projects under CNCF? anything else you would suggest.

daxmc99 replied:
This is likely fine.  There is a bit of a joke though that CNCF landscape is updated daily with new projects. Linking directly to it might only make sense if our repolist will grow dynamically. 
https://github.com/cncf/toc/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc there are new projects being added frequently

daxmc99 replied:
Perhaps use `card` mode for a cleaner look 
```suggestion
            <a href="https://landscape.cncf.io/format=card-mode&project=hosted" target="_blank" rel="noopener noreferrer">
```

poojaj-tech replied:
@daxmc99 great to know this list changes often 😅 @attfarhan I think we should remove the number to avoid having to update this continuously.
Initially we can add new projects to the repo list based on requests from the CNCF community.

sqs commented:
Looking great! A few things:

- [ ] In the ui router Go code, it would be nice if the page title while loading was `CNCF code search - Sourcegraph` not `Repogroup - Sourcegraph`. (Here, "CNCF" is OK without expanding it.) Also if the meta tags and OpenGraph tags were tailored to this CNCF page: "Search all repositories in the Cloud Native Computing Foundation (CNCF)." or something. The test will be: when you paste https://sourcegraph.com/cncf in Slack or Twitter, does the link preview look awesome/relevant/enticing?
- [ ] Remove "Query builder" link, since these users probably don't need it, and it would take them outside the CNCF scope.
- [ ] I think we should remove this notice completely. It gets in the way for CNCF, it's not super effective (afaict), and it will make less and less sense as we add support for cloud private repos. 
![image](https://user-images.githubusercontent.com/1976/94889454-1d838800-0431-11eb-8e38-560cd3cf716a.png)

attfarhan commented:
I've addressed all the feedback above, and this should be in a mergeable state. @sourcegraph/search, I'll be leaving on my holiday, could you own getting this merged?

lguychard commented on web/src/search/input/SearchPageInput.tsx:
```diff
@@ -67,6 +67,12 @@ interface Props
     globbing: boolean
     /** A query fragment to appear at the beginning of the input. */
     queryPrefix?: string
+    /** A query fragment to be appended to queries. This will not appear in the input until a search is submitted. */
```
```suggestion
    /** A query fragment to be prepended to queries. This will not appear in the input until a search is submitted. */
```

poojaj-tech commented on web/src/repogroups/cncf.tsx:
```diff
@@ -15,6 +15,6 @@ export const cncf: RepogroupMetadata = {
     ),
     examples: [],
-    homepageDescription: 'Search the 67 CNCF projects.',
+    homepageDescription: 'Search CNCF projects',
```
Description on https://sourcegraph.com/cncf still has a period but on sourcegraph.com doesn't. It should be switched.
